ts/src/test/Exchange/base/test.market.ts

```diff
@@ -102,23 +102,53 @@ function testMarket (exchange, skippedProperties, method, market) {
     const contractSize = exchange.safeString (market, 'contractSize');
     // contract fields
     if (market['contract']) {
+        // spot should be false
+        assert (!market['spot'], '"spot" must be false when "contract" is true' + logText);
+        // todo: expand logic on other market types, like "options", "index" etc.
+        // assert (isSwapOrFuture, 'either swap or future needs to be true when "contract" is true' + logText);
         // linear & inverse should have different values (true/false)
-        // todo: expand logic on other market types
         if (isSwapOrFuture) {
-            assert (market['linear'] !== market['inverse'], 'market linear and inverse must not be the same' + logText);
-            if (!('contractSize' in skippedProperties)) {
-                // contract size should be defined
-                assert (contractSize !== undefined, '"contractSize" must be defined when "contract" is true' + logText);
-                // contract size should be above zero
-                assert (Precise.stringGt (contractSize, '0'), '"contractSize" must be > 0 when "contract" is true' + logText);
-            }
-            if (!('settle' in skippedProperties)) {
-                // settle should be defined
-                assert ((market['settle'] !== undefined) && (market['settleId'] !== undefined), '"settle" & "settleId" must be defined when "contract" is true' + logText);
+            if (!('subType' in skippedProperties)) {
+                // todo:
+                if (!('quanto' in market)) {
+                    assert (market['linear'] !== market['inverse'], 'linear and inverse must not be the same' + logText);
+                } else {
+                    const linear = exchange.safeValue (market, 'linear');
+                    const inverse = exchange.safeValue (market, 'inverse');
+                    const quanto = exchange.safeValue (market, 'quanto');
+                    assert ((linear !== undefined) && (inverse !== undefined) && (quanto !== undefined), '"linear", "inverse" and "quanto" must be defined when "contract" is true' + logText);
```

Comment:
so, if this is "swap" or "future" (we are inside that condition block here), those values **must** be set (otherwise, as an exclusionary crappy exchange we add that into skip with `subType` key.